quarkus/dishmanagement/src/main/java/com/devonfw/application/dishmanagement/logic/UcDishManagement.java

```diff
@@ -0,0 +1,212 @@
+
+  @Inject
+  CategoryMapper categoryMapper;
+
+  @Inject
+  DishRepository dishRepository;
+
+  @Inject
+  DishMapper dishMapper;
+
+  @Inject
+  IngredientRepository ingredientRepository;
+
+  @Inject
+  IngredientMapper ingredientMapper;
+
+  public CategoryEto findCategory(Long id) {
+
+    Optional<CategoryEntity> entity = this.categoryRepo.findById(id);
+    return this.categoryMapper.mapTo(entity.get());
+  }
+
+  public Page<CategoryEto> findCategoryEtos(CategorySearchCriteriaTo criteria) {
+
+    Page<CategoryEto> pagListTo = null;
+    Page<CategoryEntity> categories = this.categoryRepo.findCategorys(criteria);
+    List<CategoryEto> etos = this.categoryMapper.mapList(categories.getContent());
+    if (etos.size() > 0) {
+      Pageable pagResultTo = PageRequest.of(criteria.getPageable().getPageNumber(), etos.size());
+      pagListTo = new PageImpl<>(etos, pagResultTo, pagResultTo.getPageSize());
+    }
+    return pagListTo;
+  }
+
+  public boolean deleteCategory(Long categoryId) {
+
+    this.categoryRepo.deleteById(categoryId);
+    LOG.info("Entity with Id " + categoryId + " is deleted ");
+    return true;
+  }
+
+  public CategoryEto saveCategory(CategoryEto category) {
+
+    CategoryEntity entity = this.categoryRepo.save(this.categoryMapper.mapTo(category));
+
+    return this.categoryMapper.mapTo(entity);
+  }
+
+  public DishCto findDish(Long id) {
+
+    Optional<DishEntity> dishEntity = this.dishRepository.findById(id);
+    DishEntity entity = dishEntity.get();
+    DishCto cto = new DishCto();
+    if (entity != null) {
+      cto.setDish(this.dishMapper.mapTo(entity));
+      cto.setCategories(this.categoryMapper.mapList(entity.getCategories()));
+      cto.setExtras(this.ingredientMapper.mapList(entity.getExtras()));
+    }
+
+    return cto;
+  }
+
+  public Page<DishCto> findDishCtos(DishSearchCriteriaTo criteria) {
+
+    Page<DishCto> pagListTo = null;
+    Page<DishEntity> entities = this.dishRepository.findDishs(criteria);
+    List<DishEntity> dishes = entities.getContent();
+    List<DishCto> ctos = new ArrayList<>(dishes.size());
+    for (DishEntity dish : dishes) {
+      DishCto dishCto = new DishCto();
+      dishCto.setDish(this.dishMapper.mapTo(dish));
+      dishCto.setCategories(this.categoryMapper.mapList(dish.getCategories()));
+      dishCto.setExtras(this.ingredientMapper.mapList(dish.getExtras()));
+      ctos.add(dishCto);
+    }
+    if (!criteria.getCategories().isEmpty()) {
+      ctos = categoryFilter(ctos, criteria.getCategories());
+    }
+    if (ctos.size() > 0) {
+      Pageable pagResultTo = PageRequest.of(criteria.getPageable().getPageNumber(), ctos.size());
+      pagListTo = new PageImpl<>(ctos, pagResultTo, pagResultTo.getPageSize());
+    }
+    return pagListTo;
+  }
+
+  private List<DishCto> categoryFilter(List<DishCto> dishes, List<CategoryEto> categories) {
+
+    List<DishCto> dishFiltered = new ArrayList<>();
+    for (DishCto dish : dishes) {
+
```

Comment:
Please try to avoid looping through both lists looking for matches. Use Stream API instead